### ISSUE TITLE
Add support for --exclude flag in COPY and ADD instructions

### ISSUE THREAD

**Issue by wonderfulspam** [hadolint/hadolint#1029]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Hadolint should be able to lint a Dockerfile which uses the experimental `--exclude` flag given that the Docker image builds successfully.

### Actual behavior

Hadolint fails to lint the file, producing this error:

```
invalid flag: --exclude
```

### Steps to reproduce the behavior

Run hadolint on the Dockerfile provided below.

**Output of `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.1
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:1.7-labs
FROM alpine
COPY --exclude=code/sensitive_data ./code code
```

**Additional environment details (OS, stack version, etc.)**

n/a

**Comment by mcshaman** [comment]
I too want this

**Comment by Guts** [comment]
+1 🙏

**Comment by Guts** [comment]
I've tried with latest official Docker image (Hub and GHCR) and on the online demo but the @eexit PR is still not included:

![Image](https://github.com/user-attachments/assets/218876c3-a0f2-48ce-acfa-4dd5d01e2480)

**Comment by m-ildefons** [comment]
Hi,

this is now fixed in Hadolint v2.13.1.
Unfortunately, I can't update the online version right now, but I hope to be able to fix that in the future.
